src/sparse_blas/backends/mklcpu/mklcpu_basic.cpp

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+* Copyright 2023 Codeplay Software Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+// MKLCPU backend includes
+// These includes define their own oneapi::mkl::sparse namespace with some of the types that are used here: matrix_handle_t, index_base.
+#include <oneapi/mkl/spblas.hpp>
+
+#include "oneapi/mkl/sparse_blas/detail/helper_types.hpp"
+#include "oneapi/mkl/sparse_blas/detail/mklcpu/onemkl_sparse_blas_mklcpu.hpp"
+
+// Include are set up so that oneapi::mkl::sparse namespace refers to the MKLCPU namespace in this file.
+// oneapi::mkl::sparse::detail namespace refers to the oneMKL interface namespace.
+
+namespace oneapi::mkl::sparse::detail {
+
+auto get_handle(detail::matrix_handle **handle) {
+    return reinterpret_cast<matrix_handle_t *>(handle);
+}
+auto get_handle(detail::matrix_handle *handle) {
+    return reinterpret_cast<matrix_handle_t>(handle);
+}
+
+} // namespace oneapi::mkl::sparse::detail
+
+namespace oneapi::mkl::sparse::mklcpu {
+
+void init_matrix_handle(sycl::queue & /*queue*/, detail::matrix_handle **handle) {
+    oneapi::mkl::sparse::init_matrix_handle(detail::get_handle(handle));
+}
+
+sycl::event release_matrix_handle(sycl::queue &queue, detail::matrix_handle **handle,
+                                  const std::vector<sycl::event> &dependencies) {
+    return oneapi::mkl::sparse::release_matrix_handle(queue, detail::get_handle(handle),
+                                                      dependencies);
+}
+
+template <typename fpType, typename intType>
+std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>> set_csr_data(
```

Comment:
Shouldn't this be `std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>, void>`, like you have `std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>, sycl::event>` a few lines below?